addons/web/static/src/core/utils/strings.js

```diff
@@ -1,5 +1,23 @@
 /** @odoo-module **/
 
+/**
+ * Escapes a string for HTML.
+ *
+ * @param {string | number} [str] the string to escape
+ * @returns an escaped string
+ */
+export function escape(str) {
+    if (str === undefined) {
+        return "";
+    }
+    if (typeof str === "number") {
+        return String(str);
+    }
+    const p = document.createElement("p");
+    p.textContent = str;
+    return p.innerHTML;
+}
```

Comment:
Note: this is effectively escaping a string only for using it as a node content, it doesn't escape it for using it in an attribute.

It's not new, the Owl method did that too. But it can be misleading and the nuance is really important. This method is unsafe to use when injecting in an attribute and it is not mentioned anywhere. I think it would be appropriate to mention it now.

And if we broaden the question, do we actually need such a method now that we have `t-out` and `markup`?
Note that this makes me thinking, is markup actually able to properly nuance escaping for node and for attribute? Probably not considering its API? I guess it just does it the `_` way of escaping everything in both cases, leading to being extra safe but actually escaping too much in each case (but it's better that way).

Reply:
`escape` still makes sense with `t-out` and `markup` as you may want to insert an unsafe content inside html tags. We do it somewhere in this PR, but I don't remember where. We can make escape work for attributes as well, but we'll do it later on, as it's not in the scope of this PR and we would like to merge it asap.